src/events.ts

```diff
@@ -0,0 +1,28 @@
+import {EventEmitter} from 'events'
+import {SanityDocument} from '@sanity/types'
+
+/** @public */
+export type Events = {
+  /**
+   * Emitted after the dataset was loaded.
+   */
+  datasetLoaded: {
+    dataset: string
```

Comment:
What's the purpose of this field? It seems a bit artificial to have `dataset` (and not `projectId`) here. And whenever you set up the listener you probably already know which `dataset`/`projectId` it's applied to?